From b4f52762aab3c7f39ad151afc1359c9dfbf59978 Mon Sep 17 00:00:00 2001
From: Gwendal Grignou <gwendal@chromium.org>
Date: Wed, 5 Jun 2019 16:20:20 -0700
Subject: [PATCH] CHROMIUM: mfd: cros_ec: Remove chromium specific changes in
 cros_ec_commands.h

This CL removes all changes that have not be sent upstream.

It put cros_ec_command.sh to 93abf68b6bcb git://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git master]
That the last CL upstream before the sync-up.

BUG=chromium:945948
TEST=Prepare for next CLs.
CQ-DEPEND=CL:1652467

Change-Id: Iff3392018f7cf1d7f8d2911bbf159751b53dcdf5
Signed-off-by: Gwendal Grignou <gwendal@chromium.org>
Reviewed-on: https://chromium-review.googlesource.com/1646943
Commit-Ready: ChromeOS CL Exonerator Bot <chromiumos-cl-exonerator@appspot.gserviceaccount.com>
Legacy-Commit-Queue: Commit Bot <commit-bot@chromium.org>
Reviewed-by: Benson Leung <bleung@google.com>

[rebase419(rrangel): Most things were already changed. Ther were only 3
changes that needed to come fro this  patch.]
Signed-off-by: Raul E Rangel <rrangel@chromium.org>
---
 include/linux/mfd/cros_ec_commands.h | 19 ++++++++++++-------
 1 file changed, 12 insertions(+), 7 deletions(-)

diff --git a/include/linux/mfd/cros_ec_commands.h b/include/linux/mfd/cros_ec_commands.h
index ba639f5e7d4d..fa397722f17e 100644
--- a/include/linux/mfd/cros_ec_commands.h
+++ b/include/linux/mfd/cros_ec_commands.h
@@ -3375,13 +3375,6 @@ enum ec_mkbp_event {
 
 	/* Number of MKBP events */
 	EC_MKBP_EVENT_COUNT,
-
-	/*
-	 * Maximum possible event type
-	 * The most significant bit of event type is used to indicate that
-	 * the EC has multiple events for the AP to serve
-	 */
-	EC_MKBP_EVENT_MAX_TYPE = EC_MKBP_EVENT_TYPE_MASK,
 };
 BUILD_ASSERT(EC_MKBP_EVENT_COUNT <= EC_MKBP_EVENT_TYPE_MASK);
 
@@ -3460,6 +3453,7 @@ struct ec_response_get_next_event_v1 {
 /* Switches */
 #define EC_MKBP_LID_OPEN	0
 #define EC_MKBP_TABLET_MODE	1
+#define EC_MKBP_BASE_ATTACHED	2
 
 /* Run keyboard factory test scanning */
 #define EC_CMD_KEYBOARD_FACTORY_TEST 0x0068
@@ -4831,6 +4825,17 @@ struct ec_response_usb_pd_power_info {
 	uint32_t max_power;
 } __ec_align4;
 
+
+/*
+ * This command will return the number of USB PD charge port + the number
+ * of dedicated port present.
+ * EC_CMD_USB_PD_PORTS does NOT include the dedicated ports
+ */
+#define EC_CMD_CHARGE_PORT_COUNT 0x0105
+struct ec_response_charge_port_count {
+	uint8_t port_count;
+} __ec_align1;
+
 /* Write USB-PD device FW */
 #define EC_CMD_USB_PD_FW_UPDATE 0x0110
 
-- 
2.23.0.rc1.153.gdeed80330f-goog

